hecuba_core/src/SpaceFillingCurve.cpp

```diff
@@ -273,13 +281,7 @@ int32_t ZorderCurveGenerator::computeNextClusterId() {
     std::vector<uint32_t> block_ccs = getIndexes(block_counter, blocks_dim);
     uint64_t zorder_id = computeZorder(block_ccs);
 
-    uint64_t blocks_inside_cluster = 1;
-    for (auto dim = 2; dim < blocks_dim.size(); ++dim) {
-        blocks_inside_cluster *= blocks_dim[dim];
-    }
-    uint64_t blocks_to_add = (((blocks_inside_cluster - 1) - block_counter == 0) *
-                              ((blocks_inside_cluster - 1) * (CLUSTER_SIZE << 1)));
-    block_counter += blocks_to_add > 0 ? blocks_to_add : 1;
+    block_counter += ((blocks_dim.size() == 2) * (CLUSTER_SIZE << 1)) + (blocks_dim.size() != 2);
```

Comment:
For now this line works, ill think another solution which can increase the performance

Reply:
This can be simplified to:
```
if blocks_dim.size() == 2 => add cluster<<1
else add 1

=>
block_counter += block_dim.size()==2 ? cluster_size<< 1 : 1
```